pycbc/events/stat.py

```diff
@@ -1779,8 +1787,16 @@ def single(self, trigs):
             The array of single detector values
         """
         from pycbc.conversions import mchirp_from_mass1_mass2
-        self.curr_mchirp = mchirp_from_mass1_mass2(trigs.param['mass1'],
-                                                   trigs.param['mass2'])
+        try:
+            self.curr_mchirp = mchirp_from_mass1_mass2(trigs.param['mass1'],
+                                                       trigs.param['mass2'])
+        except AttributeError:
+            try:
+                self.curr_mchirp = mchirp_from_mass1_mass2(trigs['mass1'],
+                                                           trigs['mass2'])
+            except KeyError as e:
+                print(e)
```

Comment:
Separate the operations that can actually raise the expected exception from those which do not:
```suggestion
        try:
            mass1 = trigs.param['mass1']
            mass2 = trigs.param['mass2']
        except AttributeError:
            mass1 = trigs['mass1']
            mass2 = trigs['mass2']
        self.curr_mchirp = mchirp_from_mass1_mass2(mass1, mass2)
```

Reply:
That is a much better solution, especially for code readability. Thank you, I will implement it.